levels.go

```diff
@@ -560,7 +642,77 @@ func (s *levelsController) fillTables(cd *compactDef) bool {
 	return false
 }
 
-func (s *levelsController) runCompactDef(l int, cd compactDef, limiter *rate.Limiter) (err error) {
+func (s *levelsController) computeNumSubCompaction(ranges []compactRange, size int) int {
+	n := s.kv.opt.MaxSubCompaction
+	if len(ranges) < n {
+		n = len(ranges)
+	}
+	maxOutPutFiles := int(float32(size) / (4.0 / 5.0) / float32(s.kv.opt.MaxTableSize))
+	if maxOutPutFiles == 0 {
+		maxOutPutFiles = 1
+	}
+	if maxOutPutFiles < n {
+		n = maxOutPutFiles
+	}
+	return n
+}
+
+func (s *levelsController) runSubCompacts(l int, cd compactDef, limiter *rate.Limiter) ([]*table.Table, error) {
+	type jobResult struct {
+		tbls []*table.Table
+		err  error
+	}
+
+	subRanges, totalSize := cd.computeSubCompactionRanges()
```

Comment:
By reading the function name `computeSubCompactionRanges`, I would expect subRanges` is the final slice of sub compaction ranges.
I think it would be better if we put all the computation into this function.